R/helpers.R

```diff
@@ -298,3 +301,50 @@ create_dummy_enumeratedDomain_dataframe <- function(factors) {
 
   enumeratedDomains
 }
+
+
+#' Get system metadata for all objects in a data package
+#'
+#' This is a wrapper function around `dataone::getSystemMetadata` that retrieves
+#' the system metadata for all objects in a data package and returns them as a list.
+#' It is useful for inspecting system metadata and identifying changes where needed.
+#'
+#' @param node (MNode/CNode) The Coordinating/Member Node to run the query on
+#' @param pid (character) The resource map PID of the package
+#'
+#' @return (list) A structured list of system metadata
+#'
+#' @examples
+#'\dontrun{
+#' # Set environment
+#' cn_staging <- CNode("STAGING2")
+#' knb_test <- getMNode(cn_staging, "urn:node:mnTestKNB")
+#' rm_pid <- "resource_map_urn:uuid:3e5307c4-0bf3-4fd3-939c-112d4d11e8a1"
+#'
+#' all <- get_all_sysmeta(knb_test, rm_pid)
+#'
+#' # View in RStudio data viewer to inspect
+#' View(all)
+#'
+#' # Print specific elements to console
+#' all[["Metadata"]]@rightsHolder
+#'
+#' # Create separate object
+#' sysmeta_md <- all[["Metadata"]]
+#' }
+get_all_sysmeta <- function(node, pid) {
```

Comment:
Could this be improved by adding an optional `child_packages = FALSE` argument?  In some cases there could be so many children you'd want to omit them, but if you're truly getting all the sysmeta from a package you might want to include these if they exist.

Reply:
Yeah, I think that would be a good option to have. I'll see if I can add that in easily.